src/Microsoft.AspNet.Mvc.ViewFeatures/ViewExecutor.cs

```diff
@@ -36,7 +36,20 @@ public static class ViewExecutor
         {
             var response = actionContext.HttpContext.Response;
 
-            contentType = contentType ?? DefaultContentType;
+            if (contentType == null)
+            {
+                contentType = DefaultContentType;
+
+                if (!string.IsNullOrEmpty(response.ContentType))
+                {
+                    MediaTypeHeaderValue parsedContentType;
+                    if (MediaTypeHeaderValue.TryParse(response.ContentType, out parsedContentType))
+                    {
+                        contentType = parsedContentType;
+                    }
```

Comment:
We still need to parse it. When the user specifies the encoding along with the content type, it needs to be used by the writer here. https://github.com/aspnet/Mvc/pull/3009/files#diff-62f5af455fe6d271af2d9522b4841df9L49